File: src/main/java/io/vproxy/vfx/util/imagewrapper/ImageBox.java
package io.vproxy.vfx.util.imagewrapper;

import javafx.scene.image.Image;

public interface ImageBox {
    int getWidth();

    int getHeight();

    int getRGB(int x, int y);

    CanvasBox createGraphics();

    Image toFXImage();
}

File: src-ts/src/util/imagewrapper/ImageBox.ts
// getRGB returns 0xAARRGGBB to match Java's convention.

import type { CanvasBox } from './CanvasBox.js';
import type { FXImage } from '../../javafx/ImageView.js';

export interface ImageBox {
  getWidth(): number;

  getHeight(): number;

  getRGB(x: number, y: number): number;

  createGraphics(): CanvasBox;

  toFXImage(): FXImage;
}
